distribution/push_v2.go

```diff
@@ -186,6 +187,18 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, id
 			return err
 		}
 
+		// This is a temporary environment variables used in CI to allow pushing
+		// manifest v2 schema 1 images to test-registries used for testing *pulling*
+		// these images.
+		if os.Getenv("DOCKER_ALLOW_SCHEMA1_PUSH_DONOTUSE") == "" {
+			if err.Error() == "tag invalid" {
```

Comment:
Why oh why would it fail with such a misleading error message 😞

Reply:
No idea 😞 this is what the test returned as error, so it's a "best effort"